src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/DoNotDirectlyAwaitATask.cs

```diff
@@ -78,22 +78,29 @@ public override void Initialize(AnalysisContext analysisContext)
 
         private static void AnalyzeOperation(OperationAnalysisContext context, ImmutableArray<INamedTypeSymbol> taskTypes)
         {
-            IAwaitOperation awaitExpression = context.Operation as IAwaitOperation;
+            var awaitExpression = (IAwaitOperation)context.Operation;
 
             // Get the type of the expression being awaited and check it's a task type.
-            ITypeSymbol typeOfAwaitedExpression = awaitExpression?.Operation?.Type;
+            ITypeSymbol? typeOfAwaitedExpression = awaitExpression.Operation.Type;
             if (typeOfAwaitedExpression != null && taskTypes.Contains(typeOfAwaitedExpression.OriginalDefinition))
             {
                 context.ReportDiagnostic(awaitExpression.Operation.Syntax.CreateDiagnostic(Rule));
             }
         }
 
-        private static ImmutableArray<INamedTypeSymbol> GetTaskTypes(Compilation compilation)
+        private static bool TryGetTaskTypes(Compilation compilation, [NotNullWhen(returnValue: true)] out ImmutableArray<INamedTypeSymbol> taskTypes)
```

Comment:
The return type is a value type, so it's never null.

```suggestion
        private static bool TryGetTaskTypes(Compilation compilation, out ImmutableArray<INamedTypeSymbol> taskTypes)
```

Reply:
Hmm I think it would be useful for clients to know if they need to check `IsDefault` based on this attribute.

Reply:
https://github.com/dotnet/csharplang/issues/146